packages/eslint-plugin/src/rules/consistent-type-assertions.ts

```diff
@@ -262,22 +305,57 @@ export default createRule<Options, MessageIds>({
       }
     }
 
+    function checkExpressionForArrayAssertion(
+      node: TSESTree.TSAsExpression | TSESTree.TSTypeAssertion,
+    ): void {
+      if (
+        options.assertionStyle === 'never' ||
+        options.arrayLiteralTypeAssertions === 'allow' ||
+        node.expression.type !== AST_NODE_TYPES.ArrayExpression
+      ) {
+        return;
+      }
+
+      if (
+        options.arrayLiteralTypeAssertions === 'allow-as-parameter' &&
+        isAsParameter(node)
```

Comment:
[Testing]: This can be removed and no tests fail (`&& isAsParameter(node)`), I think it's important this would have a test that covers it.